axelrod/tests/unit/test_strategy_transformers.py

```diff
@@ -101,38 +248,67 @@ def test_dual_wsls_transformer(self):
+        self.assertEqual(p1.defections, p2.defections)
+        self.assertEqual(p1.state_distribution, p2.state_distribution)
 
-    def test_dual_tft_transformer(self):
+    def test_dual_transformer_with_all_strategies(self):
         """Tests that DualTransformer produces the opposite results when faced
         with the same opponent history.
         """
-        p1 = axelrod.TitForTat()
-        p2 = DualTransformer()(axelrod.TitForTat)()
-        p3 = axelrod.CyclerCCD()  # Cycles 'CCD'
+        for s in axelrod.strategies:
+            self.assert_dual_wrapper_correct(s)
 
-        for _ in range(10):
-            p1.play(p3)
+    def test_dual_jossann_regression_test(self):
+        player_class = JossAnnTransformer((0.2, 0.3))(axelrod.Alternator)
+        self.assert_dual_wrapper_correct(player_class)
 
-        p3.reset()
-        for _ in range(10):
-            p2.play(p3)
+        player_class = JossAnnTransformer((0.5, 0.4))(axelrod.EvolvedLookerUp2_2_2)
+        self.assert_dual_wrapper_correct(player_class)
 
-        self.assertEqual(p1.history, [x.flip() for x in p2.history])
+    def test_dual_transformer_simple_play_regression_test(self):
+        """DualTransformer has failed when there were multiple DualTransformers.
+        It has also failed when DualTransformer was not the outermost
+        transformer or when other transformers were between multiple
+        DualTransformers."""
+        multiple_dual_transformers = DualTransformer()(FlipTransformer()(DualTransformer()(axelrod.Cooperator)))()
 
-    def test_dual_majority_transformer(self):
-        """Tests that DualTransformer produces the opposite results when faced
-        with the same opponent history.
-        """
-        p1 = axelrod.GoByMajority()
-        p2 = DualTransformer()(axelrod.GoByMajority)()
-        p3 = axelrod.Cycler(cycle='CDD')  # Cycles 'CDD'
+        dual_transformer_not_first = IdentityTransformer()(DualTransformer()(axelrod.Cooperator))()
 
-        for _ in range(10):
+        for _ in range(3):
+            multiple_dual_transformers.play(dual_transformer_not_first)
+
+        self.assertEqual(multiple_dual_transformers.history, [D, D, D])
+        self.assertEqual(dual_transformer_not_first.history, [D, D, D])
+
+    def test_dual_transformer_multiple_interspersed_regression_test(self):
+        """DualTransformer has failed when there were multiple DualTransformers.
+        It has also failed when DualTransformer was not the outermost
+        transformer or when other transformers were between multiple
+        DualTransformers."""
+        dual_not_first_transformer = IdentityTransformer()(DualTransformer()(axelrod.EvolvedANN))
+        self.assert_dual_wrapper_correct(dual_not_first_transformer)
+
+        multiple_dual_transformers = DualTransformer()(DualTransformer()(axelrod.WinStayLoseShift))
+        self.assert_dual_wrapper_correct(multiple_dual_transformers)
+
+    def assert_dual_wrapper_correct(self, player_class):
```

Comment:
i'd love to take credit, but my contribution to this particular test was pulling it out of the original testing code to make a function and adding `axl.seed(0)`. whoever wrote the original (`test_dual_somestrategy`) made a nice concise test that showed me exactly what `dual_transformer` needed to do.  if that was you, @theref , thanks!